src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -265,6 +267,111 @@ internal static IEnumerable<IEdmOperation> FindBestOverloadBasedOnParameters(thi
             return exactMatches.Count() > 0 ? exactMatches : functions;
         }
 
+        /// <summary>
+        /// Filters the operations by parameter names.
+        /// </summary>
+        /// <param name="functions">The operations.</param>
+        /// <param name="parameters">The list of non-binding parameter names to match.</param>
+        /// <param name="caseInsensitive">Whether case insensitive.</param>
+        /// <returns>The best matching operations based on parameters.</returns>
+        internal static IList<IEdmOperation> FindBestOverloadBasedOnParameters(this IList<IEdmOperation> functions, IList<string> parameters, bool caseInsensitive = false)
+        {
+            IList<IEdmOperation> exactMatches = new List<IEdmOperation>();
+
+            // The best match out of a list of candidates is the one that has the same number of (non-binding) parameters as specified.
+            for(int i=0; i < functions.Count; i++)
+            {
+                if (functions[i].Parameters.Count() == parameters.Count + (functions[i].IsBound ? 1 : 0))

```

Comment:
IEnumerable does not have a count property. A cast to List will also be expensive. Am I missing something here? 

---
In reply to: [352772232](https://github.com/OData/odata.net/pull/1602#discussion_r352772232) [](ancestors = 352772232)

Reply:
Sorry. here i mean the perf about the "[i]" on the IList<T>.



---
In reply to: [353939228](https://github.com/OData/odata.net/pull/1602#discussion_r353939228) [](ancestors = 353939228,352772232)

Reply:
It is random access. underlying structure for List is an array. Since this is internally used, we know that IList is implemented by an array based backing field. 

---
In reply to: [363453414](https://github.com/OData/odata.net/pull/1602#discussion_r363453414) [](ancestors = 363453414,353939228,352772232)